### ISSUE TITLE
Add SSR compatibility

### ISSUE THREAD

**Issue by eele94** [codemonk-digital/vue-responsiveness#12]
**Is your feature request related to a problem? Please describe.**
Problem Inertia SSR is not working with this plugin
ReferenceError: window is not defined

**Describe the solution you'd like**
A check whether the window is defined or not

**Comment by andrei-gheorghiu** [comment]
@eele94, I'm not very experienced with SSR, but my understanding is that with SSR the app runs twice: 
 - once on server, where there's no `window` or `document` defined
 - once on client
 
 So wrapping the entire `install()` function contents (which internally uses `window.matchMedia()`) into an if close:
 ```ts
install() {
  if (typeof window !== "undefined") {
   // code here
  }
}
```
would be enough. 

Could you confirm this?
If that's the case, I could push a SSR compatible version in a few hours.

**Comment by andrei-gheorghiu** [comment]
@eele94 Could you confirm the issue is resolved when using `v0.1.1`?

**Comment by eele94** [comment]
Thanks! Yes now it's working as expected!

I still need to do a manual SSR check before using any of the $matches functionality, but that's alright.

`_isSSR() {
      return typeof window === "undefined";
    },`

`tr v-if="!_isSSR() && $matches.isMax('sm')">`

Thanks for the quick help!

**Comment by andrei-gheorghiu** [comment]
@eele94 

> I still need to do a manual SSR check before using any of the $matches functionality, but that's alright.

You're right, I didn't think of that.
I've released a new version ([v0.1.2](https://www.npmjs.com/package/vue-responsiveness/v/0.1.2)) which shouldn't need SSR guards on `$matches` anymore. 

Again, I'm no SSR expert, but what doesn't click for me is why wouldn't one provide a dom mock (like jsdom) in SSR mode, which would kill all birds with one stone: all plugins would be instantly compatible with SSR. 
___
> Thanks for the quick help!

Most welcome.

**Comment by eele94** [comment]
Great:) I'm also no SSR expert and have been trying to many hacks and those got me in more problems than I could take care.
Maybe mocking the dom would lead to extra unneeded load and maybe unwanted results for the SSR rendering or state after the components have been hydrated. I'm also not a fan of guarding everything that works with the dom, but I'm happy it's working so far..

**Comment by Pyrodash** [comment]
Hi, this library is really good, but I am having trouble using it with Nuxt 3.

I am getting the following error: `[Vue warn]: Property "$matches" was accessed during render but is not defined on instance.`

I took a look at the source code and I saw that the `$matches` property was defined regardless of the check for `typeof window !== 'undefined'`, so I looked at the distributed npm package and everything in the install function was inside the if check, which is not reflected by the code in `lib/`.

Is the version on npm up-to-date? I think that would fix the problem.
